src/tests/CMakeLists.txt

```diff
@@ -1,6 +1,9 @@
 set(TESTS unit-tests.cpp)
+set(SOURCES ../app/doctor/doctor.cpp ../app/person/person.cpp ../app/visit/visit.cpp)
```

Comment:
My lack of knowledge makes me want to understand that.
By using set(SOURCES ...) you created variable which is later used to add_executable?

Reply:
I think it is much better to:
1) Add CMakeLists.txt in every directory (doctor, person, visit) in with we define target with add_library.
2) add "add_subdirectory" in src/app/CMakeLists.txt for every class.
3) link every library in main production target src/app/CMakeLists.txt 
4) link every library in test-target of src/tests/CMakeLists.txt.

In current version we only build the classes in test target and in production code they are not even included so nobody can use them to define their code.